src/subnavbar/navitem/index.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const navItemOf = (Type) => {
+  const component = (props) => {
+    const {active, className = '', ...other} = props;
+    const elemClass = (active ? 'active' : '') + (className ? ' ' + className : '');
+    return (
+      <li>
+        <Type {...other} className={elemClass}>{props.children}</Type>
+      </li>
+    )
+  };
+  component.defaultName = 'NavItemOf' + Type;
+  component.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    active: PropTypes.bool
+  };
+  return component;
+};
+
+export const NavItem = navItemOf('a');
+NavItem.propTypes = Object.assign({ href: PropTypes.string.isRequired }, NavItem.propTypes);
+
+export default NavItem;
```

Comment:
This code feels a bit convoluted when only one type is ever used. Could we refactor to get rid of that type abstraction, or is there a plan for it?

Reply:
The type abstraction is pretty useful if you want to use something like `'react-router'` with this (my use case for this library)

You define a different `NavItem` styled in the NavItem way:
```jsx
import React from 'react';
import {Link} from 'react-router';
import {navItemOf} from 'govuk-react-components/'

export default navItemOf(Link);
```

Reply:
I see. That's fine then.